### DISCUSSION TITLE
Can I make a button navigate to a screen + open the side drawer?

### DISCUSSION THREAD

**Discussion by Grayda** [Budibase/budibase#11022] [Help] [answered]
In previous versions of Budibase I had two screens, one for `/items` to view all items, and for `/items/:id` to edit a particular item. At some point Budibase changed to using a slide-out sidebar which broke an action I had assigned to a button (Duplicate Row, then Navigate To the new row).

Is there a way I can navigate to the `/items` screen and open up the sidebar for a particular row? Or do I just need to manually create the `:id` screen like in previous versions?

If it's a manual process, any possibility of adding a way to open the side drawer? Maybe via a query string or something, like `/items?edit=<id here>`?

**Comment by ConorWebb96** [comment, marked as answer]
Hey @Grayda,

Yup you can do something like this relatively easily within Budibase. Below I've created a simple example, were on a button click it will change the page to `items/:id` and then open the side panel. (The example I'm showing is very simple and will hopefully point you int he right direction.)

1. Add the screens. `/items`, `/items/:id`
<img width="100%" alt="Screenshot 2023-06-26 at 08 35 23" src="https://github.com/Budibase/budibase/assets/126772285/fc314dcc-87cd-484a-8999-b130969ca93e">
2. Create a button on screen `items` to link to screen `items/:id` using the `Navigate to` action. (Using hardcoded link for this example.)
<img width="100%" alt="Screenshot 2023-06-26 at 08 37 26" src="https://github.com/Budibase/budibase/assets/126772285/b12e903c-123e-4fae-b4d3-56e53508c355">
3. Click on screen `items/:id`
4. Click on the define actions button on the right side of the screen.
<img width="1298" alt="Screenshot 2023-06-26 at 08 38 18" src="https://github.com/Budibase/budibase/assets/126772285/ad49a332-4df2-432c-a7eb-d5dcbbc9e559">
5. Add a open side action.
<img width="100%" alt="Screenshot 2023-06-26 at 08 43 34" src="https://github.com/Budibase/budibase/assets/126772285/7e8a64ba-b79d-45e3-946c-1a3d0df77740">
6. Choose the side panel you wish to open. \


<img width="100%" alt="Screenshot 2023-06-26 at 08 41 35" src="https://github.com/Budibase/budibase/assets/126772285/c63b4eed-656e-46ab-b87d-cfadefb5922d">


<img width="213" alt="Screenshot 2023-06-26 at 08 41 23" src="https://github.com/Budibase/budibase/assets/126772285/18aae178-43de-4602-a192-b5c6b225a530">

**Result**
![side panel open on navigate to](https://github.com/Budibase/budibase/assets/126772285/284c5086-ba10-487c-8a67-5b4e77e186f0)

This should open the side panel on page load, with the url variable setup and the form setup correctly it should then use the url variable to choose which record to display for editing purposes. If you just want this to display change without changes being made, set it to read only on the formblock/form component.



I hope this helps.
